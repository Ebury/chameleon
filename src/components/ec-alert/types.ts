export enum PropsAlertType {
  TEXT = 'text',
  DATE = 'date',
  NUMBER = 'number',
  TEL = 'tel'
}

export enum AlertType {
    ERROR = 'error', 
    INFO = 'info', 
    SUCCESS = 'success', 
    WARNING = 'warning'
}

export enum AlertEvent {
  UPDATE_OPEN = 'update:open',
  ACTION = 'action',
  CHANGE = 'change',
}

export interface AlertEvents {
  [AlertEvent.UPDATE_OPEN]: AlertProps['open']
  [AlertEvent.ACTION]: undefined
  [AlertEvent.CHANGE]: AlertProps['open']
}

export interface AlertProps {
  propsType?: PropsAlertType,
  type: AlertType,
  title: string,
  subtitle?: string,
  dismissable?: boolean,
  buttonText?: string,
  open?: boolean,
  responsive?: boolean
}
