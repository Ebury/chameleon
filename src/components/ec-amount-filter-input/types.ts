import type { DropdownItem } from '../ec-dropdown/types';

export interface ComparisonSymbolItem extends DropdownItem<string> {}

export interface AmountFilterInputModel {
  amount?: number,
  comparisonSymbol?: ComparisonSymbolItem,
}

export interface AmountFilterInputProps {
  modelValue: AmountFilterInputModel,
  locale?: string,
  isSensitive?: boolean,
  label?: string,
  note?: string,
  amountPlaceholder?: string,
  bottomNote?: string,
  isWarning?: boolean,
  warningTooltipMessage?: string,
  errorMessage?: string,
  errorTooltipMessage?: string,
  comparisonSymbolItems: ComparisonSymbolItem[],
}
