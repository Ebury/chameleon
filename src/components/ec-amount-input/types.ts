import type { InputFieldProps } from '../ec-input-field/types';

export interface AmountInputProps extends Omit<InputFieldProps, 'modelValue'> {
  modelValue?: number | string | null,
  isMasked?: boolean,
  locale?: string,
  currency?: string,
}
