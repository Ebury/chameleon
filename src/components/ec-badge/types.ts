import { AlertType as BadgeType } from '../ec-alert/types';

export { BadgeType };

export interface BadgeProps {
  type?: BadgeType,
  value: string | number,
}
