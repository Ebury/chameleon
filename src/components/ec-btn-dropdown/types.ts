import type { AnchorHTMLAttributes } from 'vue';
import type { RouteLocationRaw } from 'vue-router';

import type { DropdownItem } from '../ec-dropdown-search/types';

export interface BtnDropdownItem<TValue = string> extends DropdownItem<TValue> {
  href?: string,
  to?: string | RouteLocationRaw,
  attrs?: AnchorHTMLAttributes,
}

export interface BtnDropdownProps<TValue = string, TBtnDropdownItem extends BtnDropdownItem<TValue> = BtnDropdownItem<TValue>> {
  items?: TBtnDropdownItem[],
  isBtnDropdownDisabled?: boolean,
  buttonText: string,
  listDataTest?: string,
  href?: string,
  to?: string | RouteLocationRaw,
}
