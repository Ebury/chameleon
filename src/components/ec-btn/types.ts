import type { RouteLocationRaw } from 'vue-router';

import type { IconName } from '../ec-icon/iconNames';

export enum ButtonSize {
  Medium = 'md',
  Small = 'sm'
}

export enum ButtonCategory {
  Primary = 'primary',
  Secondary = 'secondary',
  Success = 'success',
  Error = 'error',
  Warning = 'warning'
}

export interface ButtonProps {
  size?: ButtonSize;
  href?: string;
  to?: RouteLocationRaw;
  tag?: string;
  isDisabled?: boolean;
  icon?: IconName;
  isRounded?: boolean;
  isOutline?: boolean;
  isFullWidth?: boolean;
  category: ButtonCategory;
  isReverse?: boolean;
  isLoading?: boolean;
  isSubmit?: boolean
}
