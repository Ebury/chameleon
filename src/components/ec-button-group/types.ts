export interface ButtonGroupItem<TValue> {
  id?: string | number,
  value: TValue,
  text: string,
  disabled?: boolean,
}

export interface ButtonGroupProps<TValue> {
  modelValue?: TValue,
  items: ButtonGroupItem<TValue>[],
}
