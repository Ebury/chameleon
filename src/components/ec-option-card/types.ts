import type { IconName } from '../ec-icon/types';
import type { RouteLocationRaw } from "vue-router";

export interface OptionCardProps {
  isDisabled?: boolean,
  title: string,
  caption?: string,
  iconName?: IconName,
  type?: OptionCardType,
  to?: RouteLocationRaw,
  href?: string,
}

export enum OptionCardType {
  ACCENT = 'accent',
  DANGER = 'danger',
}

export enum OptionCardEvent {
  CLICK = 'click'
}

export interface OptionCardEvents {
  [OptionCardEvent.CLICK]: undefined
}
