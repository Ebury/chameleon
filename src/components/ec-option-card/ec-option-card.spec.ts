import type { ComponentMountingOptions } from '@vue/test-utils';
import { mount } from '@vue/test-utils';

import type { CVueWrapper } from '../../../tests/utils/global';
import { IconName } from '../ec-icon/icon-names';
import EcOptionCard from './ec-option-card.vue';
import type { OptionCardProps } from './types';
import { OptionCardType } from './types';

describe('EcOptionCard', () => {
  function mountEcOptionCard(props?: OptionCardProps, mountOpts?: ComponentMountingOptions<OptionCardProps>): CVueWrapper {
    return mount(
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      EcOptionCard as any,
      {
        props: {
          title: 'Test Option Card',
          ...props,
        },
        ...mountOpts,
      },
    ) as CVueWrapper;
  }

  it('renders properly with the required title prop', () => {
    const wrapper = mountEcOptionCard();
    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders properly when an Icon is passed', () => {
    const wrapper = mountEcOptionCard({
      iconName: IconName.SimpleEye,
      title: 'Test option card with Icon',
    });
    expect(wrapper.findByDataTest('ec-option-card__icon').exists()).toBe(true);
  });

  it('renders properly when an Icon and Title and Caption is passed', () => {
    const wrapper = mountEcOptionCard({
      iconName: IconName.SimpleEye,
      title: 'Test option card with Icon and caption',
      caption: 'Test Caption',
    });
    expect(wrapper.findByDataTest('ec-option-card__icon').exists()).toBe(true);
  });

  it.each([OptionCardType.ACCENT, OptionCardType.DANGER])('should render the Option card with the classes related to the type "%s" when type prop is set', (cardType) => {
    const wrapper = mountEcOptionCard({
      iconName: IconName.SimpleInfo,
      type: cardType,
      title: 'Test title',
      caption: 'Test Caption',
    });

    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.findByDataTest('ec-option-card').classes(`ec-option-card--${cardType}`)).toBe(true);
    expect(wrapper.findByDataTest('ec-option-card__icon').classes(`ec-option-card__icon--${cardType}`)).toBe(true);
    expect(wrapper.findByDataTest('ec-option-card__caption').classes(`ec-option-card__caption--${cardType}`)).toBe(true);
  });

  it('renders the Option Card disabled if the isDisabled prop is set to true', () => {
    const wrapper = mountEcOptionCard({
      isDisabled: true,
      iconName: IconName.SimpleEye,
      title: 'Test option card with Icon and caption',
      caption: 'Test Caption',
    });

    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.findByDataTest('ec-option-card').classes('ec-option-card--disabled')).toBe(true);
    expect(wrapper.findByDataTest('ec-option-card__icon').classes('ec-option-card__icon--disabled')).toBe(true);
    expect(wrapper.findByDataTest('ec-option-card__caption').classes('ec-option-card__caption--disabled')).toBe(true);
  });

  it('renders the Option Card disabled and with the classes related to the type "accent" when type prop is set to accent and isDisabled prop is set to true', () => {
    const wrapper = mountEcOptionCard({
      isDisabled: true,
      type: OptionCardType.ACCENT,
      iconName: IconName.SimpleEye,
      title: 'Test option card with Icon and caption',
      caption: 'Test Caption',
    });

    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.findByDataTest('ec-option-card').classes('ec-option-card--disabled')).toBe(true);
    expect(wrapper.findByDataTest('ec-option-card__icon').classes('ec-option-card__icon--disabled')).toBe(true);
    expect(wrapper.findByDataTest('ec-option-card__caption').classes('ec-option-card__caption--disabled')).toBe(true);
    expect(wrapper.findByDataTest('ec-option-card').classes('ec-option-card--accent')).toBe(true);
    expect(wrapper.findByDataTest('ec-option-card__icon').classes('ec-option-card__icon--accent')).toBe(true);
    expect(wrapper.findByDataTest('ec-option-card__caption').classes('ec-option-card__caption--accent')).toBe(true);
  });

  it('renders the Option Card disabled and with the classes related to the type "danger" when type prop is set to accent and isDisabled prop is set to true', () => {
    const wrapper = mountEcOptionCard({
      isDisabled: true,
      type: OptionCardType.DANGER,
      iconName: IconName.SimpleEye,
      title: 'Test option card with Icon and caption',
      caption: 'Test Caption',
    });

    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.findByDataTest('ec-option-card').classes('ec-option-card--disabled')).toBe(true);
    expect(wrapper.findByDataTest('ec-option-card__icon').classes('ec-option-card__icon--disabled')).toBe(true);
    expect(wrapper.findByDataTest('ec-option-card__caption').classes('ec-option-card__caption--disabled')).toBe(true);
    expect(wrapper.findByDataTest('ec-option-card').classes('ec-option-card--danger')).toBe(true);
    expect(wrapper.findByDataTest('ec-option-card__icon').classes('ec-option-card__icon--danger')).toBe(false);
    expect(wrapper.findByDataTest('ec-option-card__caption').classes('ec-option-card__caption--danger')).toBe(false);
  });
});
