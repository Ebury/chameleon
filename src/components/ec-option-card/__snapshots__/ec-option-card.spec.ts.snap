// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EcOptionCard > renders properly with the required title prop 1`] = `
<div
  class="ec-option-card"
  data-test="ec-option-card"
>
  
  <span
    data-test="ec-option-card__title-img"
  >
    <!--v-if-->
    <p
      class="ec-option-card__title"
      data-test="ec-option-card__title"
    >
      Test Option Card
    </p>
  </span>
  <!--v-if-->
  
</div>
`;

exports[`EcOptionCard > renders the Option Card disabled and with the classes related to the type "accent" when type prop is set to accent and isDisabled prop is set to true 1`] = `
<div
  class="ec-option-card ec-option-card--disabled ec-option-card--accent"
  data-test="ec-option-card ec-option-card--disabled ec-option-card--accent"
>
  
  <span
    data-test="ec-option-card__title-img"
  >
    <svg
      alt="Test option card with Icon and caption"
      class="ec-icon ec-option-card__icon ec-option-card__icon--disabled ec-option-card__icon--accent"
      data-test="ec-option-card__icon"
      height="24"
      width="24"
    >
      <use
        xlink:href="#ec-simple-eye"
      />
    </svg>
    <p
      class="ec-option-card__title"
      data-test="ec-option-card__title"
    >
      Test option card with Icon and caption
    </p>
  </span>
  <p
    class="ec-option-card__caption ec-option-card__caption--disabled ec-option-card__caption--accent"
    data-test="ec-option-card__caption"
  >
    Test Caption
  </p>
  
</div>
`;

exports[`EcOptionCard > renders the Option Card disabled and with the classes related to the type "danger" when type prop is set to accent and isDisabled prop is set to true 1`] = `
<div
  class="ec-option-card ec-option-card--disabled ec-option-card--danger"
  data-test="ec-option-card ec-option-card--disabled ec-option-card--danger"
>
  
  <span
    data-test="ec-option-card__title-img"
  >
    <svg
      alt="Test option card with Icon and caption"
      class="ec-icon ec-option-card__icon ec-option-card__icon--disabled"
      data-test="ec-option-card__icon"
      height="24"
      width="24"
    >
      <use
        xlink:href="#ec-simple-eye"
      />
    </svg>
    <p
      class="ec-option-card__title"
      data-test="ec-option-card__title"
    >
      Test option card with Icon and caption
    </p>
  </span>
  <p
    class="ec-option-card__caption ec-option-card__caption--disabled"
    data-test="ec-option-card__caption"
  >
    Test Caption
  </p>
  
</div>
`;

exports[`EcOptionCard > renders the Option Card disabled if the isDisabled prop is set to true 1`] = `
<div
  class="ec-option-card ec-option-card--disabled"
  data-test="ec-option-card ec-option-card--disabled"
>
  
  <span
    data-test="ec-option-card__title-img"
  >
    <svg
      alt="Test option card with Icon and caption"
      class="ec-icon ec-option-card__icon ec-option-card__icon--disabled"
      data-test="ec-option-card__icon"
      height="24"
      width="24"
    >
      <use
        xlink:href="#ec-simple-eye"
      />
    </svg>
    <p
      class="ec-option-card__title"
      data-test="ec-option-card__title"
    >
      Test option card with Icon and caption
    </p>
  </span>
  <p
    class="ec-option-card__caption ec-option-card__caption--disabled"
    data-test="ec-option-card__caption"
  >
    Test Caption
  </p>
  
</div>
`;

exports[`EcOptionCard > renders the Option Card with slot 1`] = `
<div
  class="ec-option-card"
  data-test="ec-option-card"
>
  
  Some slot 
  <div>
    some text
  </div>
  
</div>
`;

exports[`EcOptionCard > should render the Option card with the classes related to the type "accent" when type prop is set 1`] = `
<div
  class="ec-option-card ec-option-card--accent"
  data-test="ec-option-card ec-option-card--accent"
>
  
  <span
    data-test="ec-option-card__title-img"
  >
    <svg
      alt="Test title"
      class="ec-icon ec-option-card__icon ec-option-card__icon--accent"
      data-test="ec-option-card__icon"
      height="24"
      width="24"
    >
      <use
        xlink:href="#ec-simple-info"
      />
    </svg>
    <p
      class="ec-option-card__title"
      data-test="ec-option-card__title"
    >
      Test title
    </p>
  </span>
  <p
    class="ec-option-card__caption ec-option-card__caption--accent"
    data-test="ec-option-card__caption"
  >
    Test Caption
  </p>
  
</div>
`;

exports[`EcOptionCard > should render the Option card with the classes related to the type "danger" when type prop is set 1`] = `
<div
  class="ec-option-card ec-option-card--danger"
  data-test="ec-option-card ec-option-card--danger"
>
  
  <span
    data-test="ec-option-card__title-img"
  >
    <svg
      alt="Test title"
      class="ec-icon ec-option-card__icon ec-option-card__icon--danger"
      data-test="ec-option-card__icon"
      height="24"
      width="24"
    >
      <use
        xlink:href="#ec-simple-info"
      />
    </svg>
    <p
      class="ec-option-card__title"
      data-test="ec-option-card__title"
    >
      Test title
    </p>
  </span>
  <p
    class="ec-option-card__caption ec-option-card__caption--danger"
    data-test="ec-option-card__caption"
  >
    Test Caption
  </p>
  
</div>
`;
